tests/src/gtest/ut_gtest_main.cpp

```diff
@@ -0,0 +1,111 @@
+#include <gtest/gtest.h>
+#include <ut_kvp_profile.h>  // Include the wrapper class
+
+#define KVP_VALID_TEST_NO_FILE "assets/this_does_not_exist.yaml"
+#define KVP_VALID_TEST_YAML_FILE "assets/test_kvp.yaml"
+
+// Test fixture class for setting up and tearing down the test environment.
+class UTKVPProfileTest : public ::testing::Test {
+protected:
+    void SetUp() override {
+        // Any necessary setup before each test is run.
+    }
+
+    void TearDown() override {
+        // Ensure the profile is closed after each test to clean up.
+        ut_kvp_profile_close();
+    }
+};
+
+TEST_F(UTKVPProfileTest, TestProfileOpenSuccess) {
+    const char* validFileName = KVP_VALID_TEST_YAML_FILE;
+    ut_kvp_status_t status = ut_kvp_profile_open((char*)validFileName);
+    EXPECT_EQ(status, UT_KVP_STATUS_SUCCESS);
+}
+
+TEST_F(UTKVPProfileTest, TestProfileOpenFailure) {
+    // Test with an invalid file name.
+    const char* invalidFileName = KVP_VALID_TEST_NO_FILE;
+    ut_kvp_status_t status = ut_kvp_profile_open((char*)invalidFileName);
+    EXPECT_EQ(status, UT_KVP_STATUS_FILE_OPEN_ERROR);
+
+    // Test with a null file name.
+    status = ut_kvp_profile_open(nullptr);
+    EXPECT_EQ(status, UT_KVP_STATUS_INVALID_PARAM);
+}
+
+TEST_F(UTKVPProfileTest, TestProfileClose) {
+    // Open and then close to ensure no exceptions or errors occur.
+    ut_kvp_status_t status = ut_kvp_profile_open((char*)KVP_VALID_TEST_YAML_FILE);
+    EXPECT_EQ(status, UT_KVP_STATUS_SUCCESS);
+
+    ut_kvp_profile_close();
+    // Since close doesn't return a status, we assume success if no exceptions were thrown.
+    SUCCEED();
+}
+
+TEST_F(UTKVPProfileTest, TestProfileGetInstance) {
+    ut_kvp_status_t status = ut_kvp_profile_open((char*)KVP_VALID_TEST_YAML_FILE);
+    EXPECT_EQ(status, UT_KVP_STATUS_SUCCESS);
+
+    ut_kvp_instance_t* instance = ut_kvp_profile_getInstance();
+    EXPECT_NE(instance, nullptr);
+}
+
+TEST_F(UTKVPProfileTest, TestGetBoolField) {
+    ut_kvp_status_t status = ut_kvp_profile_open((char*)KVP_VALID_TEST_YAML_FILE);
+    EXPECT_EQ(status, UT_KVP_STATUS_SUCCESS);
+
+    bool value = UT_KVP_PROFILE_GET_BOOL("decodeTest/checkBoolFalse");
+    EXPECT_FALSE(value);
+}
+
+TEST_F(UTKVPProfileTest, TestGetUInt8Field) {
+    ut_kvp_status_t status = ut_kvp_profile_open((char*)KVP_VALID_TEST_YAML_FILE);
+    EXPECT_EQ(status, UT_KVP_STATUS_SUCCESS);
+
+    uint8_t value = UT_KVP_PROFILE_GET_UINT8("decodeTest.checkUint8IsDeHex");
+    EXPECT_EQ(value, 0xde);
+}
+
+TEST_F(UTKVPProfileTest, TestGetUInt16Field) {
+    ut_kvp_status_t status = ut_kvp_profile_open((char*)KVP_VALID_TEST_YAML_FILE);
+    EXPECT_EQ(status, UT_KVP_STATUS_SUCCESS);
+
+    uint16_t value = UT_KVP_PROFILE_GET_UINT16("decodeTest.checkUint16IsDeadHex");
+    EXPECT_EQ(value, 0xdead);
+}
+
+TEST_F(UTKVPProfileTest, TestGetUInt32Field) {
+    ut_kvp_status_t status = ut_kvp_profile_open((char*)KVP_VALID_TEST_YAML_FILE);
```

Comment:
- open & closing the profile using the C++ wrappers needs to be tested, that's one suite, you only need to load one single profile per suite of tests every test here leaks a handle.


- Other suite & thereofre file,  tests are required for all types to test your C++ macros?

Reply:
done